beacon_node/lighthouse_network/src/discovery/enr.rs

```diff
@@ -145,16 +145,39 @@ pub fn create_enr_builder_from_config<T: EnrKey>(
     enable_tcp: bool,
 ) -> EnrBuilder<T> {
     let mut builder = EnrBuilder::new("v4");
-    if let Some(enr_address) = config.enr_address {
-        builder.ip(enr_address);
+    let (maybe_ipv4_address, maybe_ipv6_address) = &config.enr_address;
+
+    if let Some(ip) = maybe_ipv4_address {
+        builder.ip4(*ip);
+    }
+
+    if let Some(ip) = maybe_ipv6_address {
+        builder.ip6(*ip);
     }
-    if let Some(udp_port) = config.enr_udp_port {
+
+    if let Some(udp_port) = config.enr_udp4_port {
```

Comment:
nitpick: What do you think about a naming convention where we call them `udp4_port` and `udp6_port` for the variable names as if they were separate types to try and not confuse ourselves. I can see you do it where it is important to distinguish, I might try get into a habit to do it everywhere, even in trivial cases like this. 

I'm sure my lack of doing this previously has been a bit of a headache in this update :p